Presscotts_Mortar_Calculator_(no_envirement)_V0.2.py

```diff
@@ -246,120 +255,121 @@ def main():
         nearEL1 = charge1[range1Lower][0]
-        avgelevation1 = (farEL1 + nearEL1)/2
-    '''charge 2'''   
+        avgelevation1 = (farEL1 + nearEL1) / 2
+    '''charge 2'''
     if type(range2Upper) and type(range2Lower) == int:
         farEL2 = charge2[range2Upper][0]
         nearEL2 = charge2[range2Lower][0]
-        avgelevation2 = (farEL2 + nearEL2)/2 
+        avgelevation2 = (farEL2 + nearEL2) / 2
     '''Flight Time Calculations'''
     if charge0Truth == 1:
         nearFT0 = charge0[range0Lower][3]
         farFT0 = charge0[range0Upper][3]
-        avgFL0 = (nearFT0 + farFT0)/2
+        avgFL0 = (nearFT0 + farFT0) / 2
     if charge1Truth == 1:
         nearFT1 = charge1[range1Lower][3]
         farFT1 = charge1[range1Upper][3]
-        avgFL1 = (nearFT1 + farFT1)/2
+        avgFL1 = (nearFT1 + farFT1) / 2
     if charge2Truth == 1:
         nearFT2 = charge2[range2Lower][3]
         farFT2 = charge2[range2Upper][3]
-        avgFL2 = (nearFT2 + farFT2)/2
-            
+        avgFL2 = (nearFT2 + farFT2) / 2
     ''' The print statments '''
-    print('='*90)
-    print (f' FIRE MISSION ')
-    print (f' Azimuth: {azimuth:.2f}') 
-    print (f' Distance from Target {distance:.2f} Meters ')
-    print (f' Elevation ranges: ')
-    print('='*90)
-    if  charge0Truth == 1:
-        print (f' [ Charge 0 Near Elevation:{nearEL0}, Far Elevation:{farEL0}] \n Average Elevation for Charge 0 {avgelevation0} \n Flight Time: {avgFL0:.2f}S')
+    print('=' * 90)
+    print(f' FIRE MISSION ')
+    print(f' Azimuth: {azimuth:.2f}')
+    print(f' Distance from Target {distance:.2f} Meters ')
+    print(f' Elevation ranges: ')
+    print('=' * 90)
+    if charge0Truth == 1:
+        print(
+            f' [ Charge 0 Near Elevation:{nearEL0}, Far Elevation:{farEL0}] \n Average Elevation for Charge 0 {avgelevation0} \n Flight Time: {avgFL0:.2f}S'
+        )
     else:
         print(' Charge 0 is invalid')
-    print('='*90)
-    if  charge1Truth == 1:
-        print (f' [ Charge 1 Near Elevation:{nearEL1}, Far Elevation:{farEL1}] \n Average Elevation for Charge 1 {avgelevation1}\n Flight Time: {avgFL1:.2f}S')
+    print('=' * 90)
+    if charge1Truth == 1:
+        print(
+            f' [ Charge 1 Near Elevation:{nearEL1}, Far Elevation:{farEL1}] \n Average Elevation for Charge 1 {avgelevation1}\n Flight Time: {avgFL1:.2f}S'
+        )
     else:
         print(' Charge 1 is invalid')
-    print('='*90)
-    if  charge2Truth == 1:
-        print (f' [ Charge 2 Near Elevation:{nearEL2}, Far Elevation:{farEL2}] \n Average Elevation for Charge 2 {avgelevation2}\n Flight Time: {avgFL2:.2f}S')
+    print('=' * 90)
+    if charge2Truth == 1:
+        print(
+            f' [ Charge 2 Near Elevation:{nearEL2}, Far Elevation:{farEL2}] \n Average Elevation for Charge 2 {avgelevation2}\n Flight Time: {avgFL2:.2f}S'
+        )
     else:
         print(' Charge 2 is invalid')
-    print('='*90)
+    print('=' * 90)
 
     main()

```

Comment:
Recursive main: generally speaking we don't want to call main from main. Instead, it would be nice if we factored out the functionality of the calculator into a function and called from function from a `while` loop.
The short reason why has to do with stack space. While it's unlikely that we'll run out, each time we call main we're layering more info on top of the stack.